src/software/ai/hl/stp/play/example/example_play_fsm.cpp

```diff
@@ -0,0 +1,32 @@
+#include "software/ai/hl/stp/play/example/example_play_fsm.h"
+
+ExamplePlayFSM::ExamplePlayFSM(const TbotsProto::AiConfig &ai_config)
+    : ai_config(ai_config)
+{
+}
+
+void ExamplePlayFSM::moveToPosition(const Update &event)
+{
+    std::vector<std::shared_ptr<MoveTactic>> move_tactics(DIV_A_NUM_ROBOTS);
+    std::generate(move_tactics.begin(), move_tactics.end(),
+                  []() { return std::make_shared<MoveTactic>(); });
```

Comment:
Right now, you're creating new tactics every time `moveToPosition` is called (i.e. every time the FSM receives an `Update` event, which is pretty much 60 times a second). You should instead make this `move_tactics` vector a private field and instantiate the tactics just once in the constructor, then reuse the same tactics every time in `moveToPosition` (updating each tactic's control params like you're doing now).

Generally, you should avoid recreating tactics (unless you want to reset them) because tactics themselves are stateful.